### PR TITLE
Fix scrape timeout config checks

### PR DESCRIPTION
**Pull request by fabxc**
@brian-brazil

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -296,11 +300,25 @@ type GlobalConfig struct {
...
-	type plain GlobalConfig
-	if err := unmarshal((*plain)(c)); err != nil {
+	if err := unmarshal(c); err != nil {
 		return err
 	}
+	if c.ScrapeTimeout > c.ScrapeInterval {
+		return fmt.Errorf("global scrape timeout greater than scrape interval")
+	}
+	if c.ScrapeTimeout == 0 {
+		if c.GlobalConfig.ScrapeTimeout > c.ScrapeInterval {
```
Comment (brian-brazil) [inline config/config.go]: s/.GlobalConfig//

File: config/config.go [hunk 2]
```diff
@@ -260,14 +260,18 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	// Do global overrides and validate unique names.
 	jobNames := map[string]struct{}{}
 	for _, scfg := range c.ScrapeConfigs {
+		if scfg.ScrapeTimeout > scfg.ScrapeInterval {
```
Comment (brian-brazil) [inline config/config.go]: I'm not sure about this. If I've set a global scrape interval (which is recommended), and then want to override the timeout I have to specify both in the job config.

I'd prefer that setting the scrape interval on a per-job level is avoided, as that tends to end in over-complication.

Comment (fabxc) [inline config/config.go]: You want to remove the scrape timing options from scrape configs entirely then?

A simpler option in general would be to silently set timeouts to `min(scrape_interval, scrape_timeout)` everywhere.

Comment (brian-brazil) [inline config/config.go]: I'm fine with having the options. I'm not so fine with setting a timeout also requiring setting the same interval as is specified in the global config, as that increases the chances that over time users will think it's perfectly okay to have different scrape intervals inside the same Prometheus.

I propose we keep this logic as-is.

Comment (fabxc) [inline config/config.go]: Ah, yes, that was not intended.
Makes sense now?

File: config/config.go [hunk 3]
```diff
@@ -260,15 +260,21 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	// Do global overrides and validate unique names.
 	jobNames := map[string]struct{}{}
 	for _, scfg := range c.ScrapeConfigs {
+		// First set the correct scrape interval, then check whether the timeout
```
Comment (brian-brazil) [inline config/config.go]: This reads oddly. Either "check whether the timeout is less than that" or "check that the timeout is not greater than that".

Same below.

Comment (fabxc) [inline config/config.go]: Fixed

**Pull request merged**